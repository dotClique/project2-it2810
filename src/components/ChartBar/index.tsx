import * as React from 'react';
import {
  ValueAxis,
  BarSeries,
  Chart,
  Title,
  ArgumentAxis,
} from '@devexpress/dx-react-chart-material-ui';
import { Animation, ValueScale } from '@devexpress/dx-react-chart';
import ChartContainer from '../ChartContainer';

interface IDataItem {
  barValue: number;
  barLabel: string;
}

export default class ChartBar extends React.Component<{ data: IDataItem[]; title: string }, {}> {
  render(): React.ReactNode {
    return (
      <ChartContainer>
        <Chart data={this.props.data}>
          <ValueScale name="barValue" />
          <ArgumentAxis />
          <ValueAxis scaleName="barValue" showGrid={false} showLine={true} showTicks={true} />

          <BarSeries
            name="Units Sold"
            valueField="barValue"
            argumentField="barLabel"
            scaleName="barValue"
          />
          <Title text={this.props.title} />
          <Animation />
        </Chart>
      </ChartContainer>
    );
  }
}
