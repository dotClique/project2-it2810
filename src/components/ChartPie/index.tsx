import * as React from 'react';
import Paper from '@material-ui/core/Paper';
import { Chart, PieSeries, Title, Tooltip } from '@devexpress/dx-react-chart-material-ui';
import { Animation, EventTracker } from '@devexpress/dx-react-chart';

type ChartPieProps = {
  data: { commitType: string; val: number }[];
  title?: string;
};

export default function ChartPie(props: ChartPieProps) {
  return (
    <Paper>
      <Chart data={props.data}>
        <PieSeries valueField="val" argumentField="commitType" innerRadius={0.6} />
        <Title text={props.title} />
        <EventTracker />
        <Tooltip />
        <Animation />
      </Chart>
    </Paper>
  );
}
