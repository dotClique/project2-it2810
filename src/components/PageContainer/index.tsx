import useStyles from './styles';
import { Container, Grid } from '@material-ui/core';
import { ReactNode } from 'react';
type PageContainerProps = {
  children: ReactNode;
  title?: string;
};

export default function PageContainer(props: PageContainerProps) {
  const style = useStyles();
  return (
    <Grid container className={style.background}>
      <Container className={style.main} maxWidth="md">
        {props.title ? <h3>{props.title}</h3> : false}
        {props.children}
      </Container>
    </Grid>
  );
}
