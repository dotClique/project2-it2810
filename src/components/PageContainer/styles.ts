import { createStyles, makeStyles } from '@material-ui/core';

const useStyles = makeStyles((theme) =>
  createStyles({
    main: {
      color: theme.palette.primary.contrastText,
      backgroundColor: theme.palette.primary.main,
      borderRadius: '0 0 10px 10px',
      paddingBottom: '20px',
    },
  }),
);

export default useStyles;
