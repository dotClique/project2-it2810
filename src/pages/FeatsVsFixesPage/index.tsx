import PageContainer from '../../components/PageContainer/index';
import ChartPie from '../../components/ChartPie';
import { getAllCommitsFromAPI } from '../../helpers/api-calls';
import { useEffect, useState } from 'react';
import { CommitAuthor } from '../../helpers/types';
import { Checkbox } from '@material-ui/core';
import { parseCommitData } from './utils';

export default function FeatsVsFixesPage() {
  const [authorData, setAuthorData] = useState<CommitAuthor[]>([]);

  let featsFixesGraphData: Array<{ commitType: string; val: number }> = [
    { commitType: 'feat', val: 0 },
    { commitType: 'fix', val: 0 },
  ];

  let additionsDeletionsGraphData: Array<{ commitType: string; val: number }> = [
    { commitType: 'additions', val: 0 },
    { commitType: 'deletions', val: 0 },
  ];

  for (let i = 0; i < authorData.length; i++) {
    if (authorData[i].active) {
      featsFixesGraphData[0].val += authorData[i].feats;
      featsFixesGraphData[1].val += authorData[i].fixes;
      additionsDeletionsGraphData[0].val += authorData[i].additions;
      additionsDeletionsGraphData[1].val += authorData[i].deletions;
    }
  }
  useEffect(() => {
    getAllCommitsFromAPI().then((res) => {
      if (res) {
        setAuthorData(parseCommitData(res));
      }
    });
  }, []);

  return (
    <PageContainer title="Commit statistics">
      Here you can see data about the ratio between feats and fixes, and the ratio between lines of
      code added and deleted. This is cumulative data of all the commits in the master branch. You
      can also choose who you want to see data about. This is anonymised to preserve the
      authors&apos; privacy.
      {authorData.map((m, i) => {
        if (m.feats || m.fixes) {
          return (
            <div key={i}>
              Person {i + 1}
              <Checkbox
                checked={m.active}
                onChange={() => {
                  const temp_list = [...authorData];
                  temp_list[i].active = !temp_list[i].active;
                  setAuthorData(temp_list);
                }}
              />
            </div>
          );
        }
      })}
      <ChartPie data={featsFixesGraphData} title={'feats vs fixes'} legend />
      <ChartPie data={additionsDeletionsGraphData} title={'additions vs deletions'} legend />
    </PageContainer>
  );
}
