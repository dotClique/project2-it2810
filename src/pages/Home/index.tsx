import PageContainer from '../../components/PageContainer/index';
import reactLogo from '../../assets/react.png';
import gitlabLogo from '../../assets/gitlab.jpg';
import materialUI from '../../assets/materialui.png';
import useStyles from './styles';
export default function Home() {
  const classes = useStyles();
  return (
    <PageContainer title="Intro">
      <div>
        In this project we visualize gitlab data about the repository used when developing this
        website. We use react and material UI for most of the layout, and use devexpress graphs for
        the base graph components.
      </div>
      <div className={classes.imgContainer}>
        <img src={reactLogo} alt={'react'} />
        <img src={gitlabLogo} alt={'react'} />
        <img src={materialUI} alt={'react'} />
      </div>
    </PageContainer>
  );
}
