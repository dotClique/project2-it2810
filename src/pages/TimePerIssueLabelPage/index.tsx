import PageContainer from '../../components/PageContainer';

export default function TimePerIssueLabelPage() {
  return (
    <PageContainer title="Average Time Per Issue-label">
      <div>PageContainer</div>
    </PageContainer>
  );
}
