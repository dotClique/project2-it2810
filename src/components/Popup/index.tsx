import { XIcon } from '@heroicons/react/outline';
import { Dialog, DialogProps } from '@material-ui/core';
import IconButton from '@material-ui/core/IconButton';
import { useStyles } from './styles';
import { ReactNode } from 'react';

type PopupProps = {
  children?: ReactNode;
  open: boolean;
  onClose: () => void;
  title: string;
  maxWidth?: DialogProps['maxWidth'];
};

/**
 * A component making a popup dialog with the desired format
 */
export default function Popup(props: PopupProps) {
  const classes = useStyles();
  return (
    <Dialog
      open={props.open}
      onClose={(_event, reason) => reason !== 'backdropClick' && props.onClose()}
      fullWidth={true}
      maxWidth={props.maxWidth ?? 'xs'}
      PaperProps={{ className: classes.dialog }}
    >
      <IconButton aria-label="close" className={classes.closeButton} onClick={props.onClose}>
        <XIcon width="1em" />
      </IconButton>
      <h1 id="dialogTitle" className={classes.title}>
        {props.title}
      </h1>
      <div className={classes.contentDiv}>{props.children}</div>
    </Dialog>
  );
}
