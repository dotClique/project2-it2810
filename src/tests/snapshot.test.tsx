import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import ChartBar from '../components/ChartBar';
import ChartPie from '../components/ChartPie';
import PageContainer from '../components/PageContainer';
import Popup from '../components/Popup';
import CommitsPerBranchPage from '../pages/CommitsPerBranch/index';
import FeatsVsFixesPage from '../pages/FeatsVsFixesPage/index';
import Home from '../pages/Home';
import TimePerIssueLabelPage from '../pages/TimePerIssueLabelPage/index';
import { CHART_BAR_DATA, CHART_PIE_DATA } from './dummydata';

Enzyme.configure({ adapter: new Adapter() });

describe('Snapshot tests', () => {
  // Snapshotting some components that many other components rely on and therefore should not be changed.
  test('Snapshotting ChartBar', () => {
    const result = toJson(shallow(<ChartBar data={CHART_BAR_DATA} title="Some title" />));
    expect(result).toMatchSnapshot();
  });

  test('Snapshotting ChartPie', () => {
    const result = toJson(shallow(<ChartPie data={CHART_PIE_DATA} title="Some title" />));
    expect(result).toMatchSnapshot();
  });

  test('Snapshotting Popup', () => {
    const result = toJson(shallow(<Popup open={true} title="Popup" onClose={() => {}} />));
    expect(result).toMatchSnapshot();
  });

  test('Snapshotting PageContainer', () => {
    const result = toJson(
      shallow(
        <PageContainer title="Some title">
          <div>Hello world</div>
        </PageContainer>,
      ),
    );
    expect(result).toMatchSnapshot();
  });

  // Snapshotting the pages that are not indended to be changed
  test('Snapshotting FeatVsFixesPage', () => {
    const result = toJson(shallow(<FeatsVsFixesPage />));
    expect(result).toMatchSnapshot();
  });

  test('Snapshotting CommitsPerBranch page', () => {
    const result = toJson(shallow(<CommitsPerBranchPage />));
    expect(result).toMatchSnapshot();
  });

  test('Snapshotting Home page', () => {
    const result = toJson(shallow(<Home />));
    expect(result).toMatchSnapshot();
  });

  test('Snapshotting TimePerIssueLabelPage', () => {
    const result = toJson(shallow(<TimePerIssueLabelPage />));
    expect(result).toMatchSnapshot();
  });
});
