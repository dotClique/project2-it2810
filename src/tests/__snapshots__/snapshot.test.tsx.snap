// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests Snapshotting ChartBar 1`] = `
<ChartContainer>
  <Component
    data={
      Array [
        Object {
          "barLabel": "Label1",
          "barValue": 4,
        },
        Object {
          "barLabel": "Label2",
          "barValue": 6,
        },
      ]
    }
  >
    <Component
      name="barValue"
    />
    <Component />
    <Component
      scaleName="barValue"
      showGrid={false}
      showLine={true}
      showTicks={true}
    />
    <Component
      argumentField="barLabel"
      barWidth={0.9}
      name="Units Sold"
      scaleName="barValue"
      valueField="barValue"
    />
    <Component
      text="Some title"
    />
    <AnimationBase
      duration={1000}
      easing={[Function]}
    />
  </Component>
</ChartContainer>
`;

exports[`Snapshot tests Snapshotting ChartPie 1`] = `
<ChartContainer>
  <Component
    data={
      Array [
        Object {
          "commitType": "Label1",
          "val": 4,
        },
        Object {
          "commitType": "Label2",
          "val": 6,
        },
      ]
    }
  >
    <Component />
    <Component
      argumentField="commitType"
      innerRadius={0.6}
      outerRadius={1}
      valueField="val"
    />
    <Component
      text="Some title"
    />
    <EventTrackerBase />
    <Component />
    <AnimationBase
      duration={1000}
      easing={[Function]}
    />
  </Component>
</ChartContainer>
`;

exports[`Snapshot tests Snapshotting CommitsPerBranch page 1`] = `
<PageContainer>
  <header />
  <div
    className="makeStyles-switchcontainer-9"
  />
</PageContainer>
`;

exports[`Snapshot tests Snapshotting FeatVsFixesPage 1`] = `
<PageContainer
  title="Commit statistics"
>
  Here you can see data about the ratio between feats and fixes, and the ratio between lines of code added and deleted. This is cumulative data of all the commits in the master branch. You can also choose who you want to see data about. This is anonymised to preserve the authors' privacy.
  <ChartPie
    data={
      Array [
        Object {
          "commitType": "feat",
          "val": 0,
        },
        Object {
          "commitType": "fix",
          "val": 0,
        },
      ]
    }
    legend={true}
    title="feats vs fixes"
  />
  <ChartPie
    data={
      Array [
        Object {
          "commitType": "additions",
          "val": 0,
        },
        Object {
          "commitType": "deletions",
          "val": 0,
        },
      ]
    }
    legend={true}
    title="additions vs deletions"
  />
</PageContainer>
`;

exports[`Snapshot tests Snapshotting Home page 1`] = `
<PageContainer
  title="Intro"
>
  <div>
    In this project we visualize gitlab data about the repository used when developing this website. We use react and material UI for most of the layout, and use devexpress graphs for the base graph components.
  </div>
  <div
    className="makeStyles-imgContainer-10"
  >
    <div>
      <img
        alt="react"
        src="react.png"
      />
    </div>
    <div>
      <img
        alt="react"
        src="gitlab.png"
      />
    </div>
    <div>
      <img
        alt="react"
        className="makeStyles-materialUILogo-11"
        src="materialui.png"
      />
    </div>
  </div>
</PageContainer>
`;

exports[`Snapshot tests Snapshotting PageContainer 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-background-6"
  container={true}
>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-main-5"
    maxWidth="md"
  >
    <h3>
      Some title
    </h3>
    <div>
      Hello world
    </div>
  </WithStyles(ForwardRef(Container))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`Snapshot tests Snapshotting Popup 1`] = `
<WithStyles(ForwardRef(Dialog))
  PaperProps={
    Object {
      "className": "makeStyles-dialog-4",
    }
  }
  fullWidth={true}
  maxWidth="xs"
  onClose={[Function]}
  open={true}
>
  <WithStyles(ForwardRef(IconButton))
    aria-label="close"
    className="makeStyles-closeButton-3"
    onClick={[Function]}
  >
    <ForwardRef(XIcon)
      width="1em"
    />
  </WithStyles(ForwardRef(IconButton))>
  <h1
    className="makeStyles-title-1"
    id="dialogTitle"
  >
    Popup
  </h1>
  <div
    className="makeStyles-contentDiv-2"
  />
</WithStyles(ForwardRef(Dialog))>
`;

exports[`Snapshot tests Snapshotting TimePerIssueLabelPage 1`] = `
<PageContainer
  title="Average  Close Time Per Issue-label"
>
  <p>
    This counts only finished issues and it is the total time from the issue was created to it was closed.
  </p>
  <div>
    <WithStyles(ForwardRef(FormControl))
      className="makeStyles-dropdown-12"
    >
      <WithStyles(ForwardRef(InputLabel))>
        Showing labels
      </WithStyles(ForwardRef(InputLabel))>
      <WithStyles(ForwardRef(Select))
        className="makeStyles-select-13"
        id="demo-simple-select"
        labelId="demo-simple-select-label"
        onChange={[Function]}
        value="[\\"Good first issue\\",\\"Easy\\",\\"Hard\\",\\"Possibly impossible\\"]"
      >
        <WithStyles(ForwardRef(MenuItem))
          key="[\\"Good first issue\\",\\"Easy\\",\\"Hard\\",\\"Possibly impossible\\"]"
          value="[\\"Good first issue\\",\\"Easy\\",\\"Hard\\",\\"Possibly impossible\\"]"
        >
          <div>
            Difficulty
          </div>
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="[\\"API\\",\\"Browser storage\\",\\"DevOps\\",\\"Testing\\"]"
          value="[\\"API\\",\\"Browser storage\\",\\"DevOps\\",\\"Testing\\"]"
        >
          <div>
            Tech Description
          </div>
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="[\\"Documentation\\",\\"Doing\\",\\"Important\\",\\"Set-up\\",\\"Planning\\"]"
          value="[\\"Documentation\\",\\"Doing\\",\\"Important\\",\\"Set-up\\",\\"Planning\\"]"
        >
          <div>
            Other
          </div>
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(Select))>
    </WithStyles(ForwardRef(FormControl))>
  </div>
</PageContainer>
`;
